src/bootstrap/step.rs

```diff
@@ -814,6 +814,9 @@ invalid rule dependency graph detected, was a rule added and maybe typo'd?
             Subcommand::Clean => panic!(),
         };
 
+        // Only construct our plan from build triple steps for dist builds.
+        let from_build_only = kind == Kind::Dist;
```

Comment:
Okay, I've finally figured out something that'll make the borrow checker happy. Is it acceptable to rely on the fact that `config.rs` explicitly puts the build triple first in the hosts array? Otherwise it seems anything that involves cloning `self.build.config.build` would irritate the borrow checker. (Completely moving to iterator-based tricks would enrage the type checker instead, btw.)